diff --git a/node_modules/pbkdf2/lib/sync-browser.js b/node_modules/pbkdf2/lib/sync-browser.js
index 20a64b4..b668552 100644
--- a/node_modules/pbkdf2/lib/sync-browser.js
+++ b/node_modules/pbkdf2/lib/sync-browser.js
@@ -19,6 +19,8 @@ var sizes = {
   ripemd160: 20
 }
 
+const hmacCache = {[key: string]: string} = {}
+
 function Hmac (alg, key, saltLen) {
   var hash = getDigest(alg)
   var blocksize = (alg === 'sha512' || alg === 'sha384') ? 128 : 64
@@ -68,6 +70,11 @@ function getDigest (alg) {
 }
 
 function pbkdf2 (password, salt, iterations, keylen, digest) {
+  const cacheKey = password + salt + iterations + keylen + digest
+  if (hmacCache[cacheKey]) {
+    return hmacCache[cacheKey]
+  }
+
   checkParameters(iterations, keylen)
   password = toBuffer(password, defaultEncoding, 'Password')
   salt = toBuffer(salt, defaultEncoding, 'Salt')
@@ -99,6 +106,8 @@ function pbkdf2 (password, salt, iterations, keylen, digest) {
     destPos += hLen
   }
 
+  hmacCache[cacheKey] = DK
+
   return DK
 }
 
